Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -266,16 +273,36 @@ private string CreateSlug(EventDto eventDto)
         baseSlug = Regex.Replace(baseSlug, @"[^a-z0-9-]", "");
         baseSlug = Regex.Replace(baseSlug, "-+", "-");
 
-        var slug = baseSlug;
+        var existingSlugs = context.Events
+            .Where(e => e.Slug.StartsWith(baseSlug))
+            .Select(e => e.Slug)
+            .ToList();
+
         var counter = 1;
+        if (existingSlugs.Any())
+        {
+            var regex = new Regex($"^{Regex.Escape(baseSlug)}-(\\d+)$");
+            foreach (var slug in existingSlugs)
+            {
+                var match = regex.Match(slug);

```

Comment:
Can't you check for a match in getting from DB, wouldn't that be easier? (it would certainly be more optimal)

Reply:
Don't know if you checked the link I sent you and tried it

Reply:
changed